packages/table/src/table.scss

```diff
@@ -44,6 +44,12 @@ $menu-z-index: $row-z-index + 1 !default;
   display: flex;
   flex: 0 0 auto;
   min-height: 0%; // Must use % unit for IE11
+
+  .bp-table-container.bp-table-no-rows & {
```

Comment:
Here's the one place the new `bp-table-no-rows` class is necessary. Tried to find a way to fix styles without it, but didn't succeed. This gets the job done.